server/service/src/document/encounter/update.rs

```diff
@@ -0,0 +1,262 @@
+                    data: serde_json::to_value(&patient).unwrap(),
+                    schema_id: schema.id.clone(),
+                    parent: None,
+                },
+            )
+            .unwrap();
+        let program = SchemaProgram {
+            enrolment_datetime: Utc::now().to_rfc3339(),
+        };
+        let program_type = "ProgramType".to_string();
+        service_provider
+            .program_service
+            .upsert_program(
+                &ctx,
+                &service_provider,
+                "store_a".to_string(),
+                "user",
+                UpsertProgram {
+                    data: serde_json::to_value(program.clone()).unwrap(),
+                    schema_id: schema.id.clone(),
+                    parent: None,
+                    patient_id: patient.id.clone(),
+                    r#type: program_type.clone(),
+                },
+            )
+            .unwrap();
+        let service = &service_provider.encounter_service;
+        let encounter = SchemaEncounter {
+            encounter_datetime: Utc::now().to_rfc3339(),
+            status: None,
+        };
+        let program_type = "ProgramType".to_string();
+        let initial_encounter = service
+            .insert_encounter(
+                &ctx,
+                &service_provider,
+                "store_a".to_string(),
+                "user",
+                InsertEncounter {
+                    data: serde_json::to_value(encounter.clone()).unwrap(),
+                    schema_id: schema.id.clone(),
+                    patient_id: patient.id.clone(),
+                    r#type: program_type.clone(),
+                    program_type: program_type.clone(),
+                },
+            )
+            .unwrap();
+
+        // InvalidParentId
+        let err = service
+            .update_encounter(
+                &ctx,
+                &service_provider,
+                "store_a".to_string(),
+                "user",
+                UpdateEncounter {
+                    data: json!({"enrolment_datetime": true}),
+                    schema_id: schema.id.clone(),
+                    parent: "invalid".to_string(),
+                },
+            )
+            .err()
+            .unwrap();
+        matches!(err, UpdateEncounterError::InvalidParentId);
+
+        // InvalidDataSchema
+        let err = service
+            .update_encounter(
+                &ctx,
+                &service_provider,
+                "store_a".to_string(),
+                "user",
+                UpdateEncounter {
+                    data: json!({"encounter_datetime": true}),
+                    schema_id: schema.id.clone(),
+                    parent: initial_encounter.id.clone(),
+                },
+            )
+            .err()
+            .unwrap();
+        matches!(err, UpdateEncounterError::InvalidDataSchema(_));
+
+        // success update
+        let encounter = SchemaEncounter {
```

Comment:
Should we check that things were updated correctly ?

Reply:
I added a check that updated data matches the input, is that what you meant?

Reply:
yep thanks